### PR TITLE
feat : Notification Lab - Reward from Completing Req Test from Service Requested

### PR DESCRIPTION
**Pull request by rubenkristian**
### JIRA LINK
- https://blocksphere2020.atlassian.net/browse/DBIO-2517

### Changelog
- send notification to lab after completing request test

### CODE REVIEW

File: src/listeners/substrate-listener/commands/orders/order-fulfilled/order-fulfilled.handler.ts [hunk 1]
```diff
@@ -152,12 +152,27 @@ export class OrderFulfilledHandler
         };
         await this.rewardService.insert(dataCustomerLoggingInput);
 
+        // Write Logging Notification Lab Reward From Request Service
+        const labNotificationInput: NotificationDto = {
+          role: 'Lab',
+          entity_type: 'Reward',
+          entity: 'OrderFulfilled',
```
Comment (hilyds) [inline src/listeners/substrate-listener/commands/orders/order-fulfilled/order-fulfilled.handler.ts:159]: entity should be `'Request Service Staking '`
image

File: src/listeners/substrate-listener/commands/orders/order-fulfilled/order-fulfilled.handler.ts [hunk 2]
```diff
@@ -152,12 +152,27 @@ export class OrderFulfilledHandler
...
+          description: `Congrats! You’ve got ${servicePrice} DBIO as a reward for completing the request test for ${order.id} from the service requested`,
```
Comment (hilyds) [inline src/listeners/substrate-listener/commands/orders/order-fulfilled/order-fulfilled.handler.ts:160]: and `${order.id}` supposed to be tracking ID instead

**Review by hilyds — CHANGES_REQUESTED** [review]

File: src/listeners/substrate-listener/commands/orders/order-fulfilled/order-fulfilled.handler.ts [hunk 3]
```diff
@@ -112,28 +112,29 @@ export class OrderFulfilledHandler
         );
         const servicePrice = +order.prices[0].value * debioToDai;
```
Comment (hilyds) [inline src/listeners/substrate-listener/commands/orders/order-fulfilled/order-fulfilled.handler.ts:113]: change this to totalPrice

File: src/listeners/substrate-listener/commands/orders/order-fulfilled/order-fulfilled.handler.ts [hunk 4]
```diff
@@ -156,8 +157,25 @@ export class OrderFulfilledHandler
...
-          order.customerId,
+          order.sellerId,
           convertToDbioUnitString(servicePrice / 10),
+          () => {
+            // Write Logging Notification Lab Reward From Request Service
+            const labNotificationInput: NotificationDto = {
+              role: 'Lab',
+              entity_type: 'Reward',
+              entity: 'Request Service Staking',
+              description: `Congrats! You’ve received ${servicePrice} DBIO for completing the request test for ${order.dnaSampleTrackingId} from the service requested.`,
```
Comment (hilyds) [inline src/listeners/substrate-listener/commands/orders/order-fulfilled/order-fulfilled.handler.ts:168]: supposed to be totalPrice

File: src/listeners/substrate-listener/commands/orders/order-fulfilled/order-fulfilled.handler.ts [hunk 5]
```diff
@@ -112,28 +112,29 @@ export class OrderFulfilledHandler
...
           convertToDbioUnitString(servicePrice),
-          () =>
-            this.callbackInsertNotificationLogging(customerNotificationInput),
+          () => {
+            // Write Logging Notification Customer Reward From Request Service
+            const customerNotificationInput: NotificationDto = {
+              role: 'Customer',
+              entity_type: 'Order',
+              entity: 'OrderFulfilled',
+              description: `Congrats! You’ve got ${servicePrice} DBIO as a reward for completing the request test for ${order.id} from the service requested`,
```
Comment (hilyds) [inline src/listeners/substrate-listener/commands/orders/order-fulfilled/order-fulfilled.handler.ts:127]: supposed to be totalPrice

File: src/listeners/substrate-listener/commands/orders/order-fulfilled/order-fulfilled.handler.ts [hunk 6]
```diff
@@ -156,8 +157,25 @@ export class OrderFulfilledHandler
         await sendRewards(
           this.substrateService.api as any,
           this.substrateService.pair,
-          order.customerId,
+          order.sellerId,
           convertToDbioUnitString(servicePrice / 10),
```
Comment (hilyds) [inline src/listeners/substrate-listener/commands/orders/order-fulfilled/order-fulfilled.handler.ts:161]: supposed to be totalPrice

File: src/listeners/substrate-listener/commands/orders/order-fulfilled/order-fulfilled.handler.ts [hunk 7]
```diff
@@ -112,28 +112,29 @@ export class OrderFulfilledHandler
...
-          from: 'Debio Network',
-          to: order.customerId,
-        };
-
         // Send reward to customer
         await sendRewards(
           this.substrateService.api as any,
           this.substrateService.pair,
           order.customerId,
           convertToDbioUnitString(servicePrice),
```
Comment (hilyds) [inline src/listeners/substrate-listener/commands/orders/order-fulfilled/order-fulfilled.handler.ts:120]: supposed to be totalPrice

**Review by hilyds — APPROVED** [review]

**Review by agustinustheo — APPROVED** [review]

**Pull request merged**